common:
    lid_topic:  "/os_cloud_node/points"
    imu_topic:  "/os_cloud_node/imu"
    #imu_topic:  "/dji_sdk/imu"
    time_sync_en: false         # ONLY turn on when external time synchronization is really not possible
    
preprocess:
    lidar_type: 3                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR, 
    scan_line: 128
    timestamp_unit: 3                 # 0-second, 1-milisecond, 2-microsecond, 3-nanosecond.
    blind: 1

point_filter:
    width: 1024
    height: 128
    w_filter_size: 4
    h_filter_size: 2
    max_var_mult: 2.0

mapping:
    acc_cov: 0.1
    gyr_cov: 0.1
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    fov_degree:    360
    det_range:     55.0
    extrinsic_est_en:  true   # true: enable the online estimation of IMU-LiDAR extrinsic
    #id 
    extrinsic_T: [ 0.0, 0.0, 0.0 ]
    extrinsic_R: [ 1,  0,  0,
                   0,  1,  0,
                   0,  0,  1]
    
    # m210 - drz2
    #extrinsic_T: [ 0.0982146, 0.0, -0.210836 ]
    #extrinsic_R: [ 0, -1,  0,
    #              -1,  0,  0,
    #               0,  0, -1]
    # m210 - mbzirc8
    #extrinsic_T: [ 0.0982146, 0.031775, -0.210836 ]
    #extrinsic_R: [ 0, -1,  0,
    #              -1,  0,  0,
    #               0,  0, -1]
    #extrinsic_T: [ 0.0805585, 0.0146624, -0.214918 ] # r1
    #extrinsic_R: [ 0.0122792, -0.99989,    -0.00837646,
    #              -0.9999,    -0.0122198,  -0.00711093,
    #               0.00700779, 0.00846294, -0.99994]
    #extrinsic_T: [ 0.06769, 0.00372607, -0.218236 ] # r2
    #extrinsic_R: [ 0.015141, -0.999804,  -0.0127657,
    #              -0.999831, -0.0150053, -0.0106552,
    #               0.0104615, 0.0129248, -0.999862]
    # m300
    #extrinsic_T: [ 0.06, 0.0, 0.0 ]
    #extrinsic_R: [ -1,  0,  0,
    #                0, -1,  0,
    #                0,  0,  1]

publish:
    path_en:  false
    scan_publish_en:  true       # false: close all the point cloud output
    dense_publish_en: false       # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: false  # true: output the point cloud scans in IMU-body-frame

pcd_save:
    pcd_save_en: false
    interval: 50                # how many LiDAR frames saved in each pcd file; 
                                 # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.
